tests/test_http.py

```diff
@@ -1053,7 +1053,10 @@ def test_add_fail_for_resumable(self):
     upload = MediaFileUpload(
         datafile('small.png'), chunksize=500, resumable=True)
     self.request1.resumable = upload
-    self.assertRaises(BatchError, batch.add, self.request1, request_id='1')
+    with self.assertRaises(BatchError) as batch_error:
+      batch.add(self.request1, request_id='1')
+    self.assertEqual(str(batch_error.exception), '<BatchError "Media requests cannot be used in a batch request.">')
```

Comment:
I absolutely agree that we should test that `batch.add(self.request1, request_id='1') raises `BatchError`. I'm suggesting that we don't want to write a test that fails if we later decide to word the error message slightly differently.

Tests form part of the definition of correctness of the system under test. I'm saying that we don't want to consider our system incorrect if the message associated with the exception that is raised is different than the message it currently raises.